pkg/diff/chunk.go

```diff
@@ -423,43 +292,63 @@ func (s *bucketSpliter) getChunksByBuckets() ([]*ChunkRange, error) {
 		if !ok {
 			return nil, errors.NotFoundf("index %s in buckets info", index.Name.O)
 		}
+		log.Debug("buckets for index", zap.String("index", index.Name.O), zap.Reflect("buckets", buckets))
 
 		var (
-			lowerValues []string
-			upperValues []string
-			latestCount int64
-			err         error
+			lowerValues, upperValues []string
+			latestCount              int64
 		)
 
 		indexColumns := getColumnsFromIndex(index, s.table.info)
 
-		for i, bucket := range buckets {
-			upperValues, err = dbutil.AnalyzeValuesFromBuckets(bucket.UpperBound, indexColumns)
-			if err != nil {
-				return nil, errors.Trace(err)
+		for i := 0; i <= len(buckets); i++ {
+			if i != len(buckets) {
+				upperValues, err = dbutil.AnalyzeValuesFromBuckets(buckets[i].UpperBound, indexColumns)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
 			}
 
-			if bucket.Count-latestCount > int64(s.chunkSize) || i == len(buckets)-1 {
-				// create a new chunk
-				chunk := NewChunkRange(bucketMode)
-				var lower, upper, lowerSymbol, upperSymbol string
-				for j, col := range index.Columns {
-					if len(lowerValues) != 0 {
-						lower = lowerValues[j]
-						lowerSymbol = gt
+			chunk := NewChunkRange()
+			for j, column := range indexColumns {
+				if i == 0 {
+					chunk.update(column.Name.O, "", upperValues[j])
+				} else if i == len(buckets) {
+					if len(lowerValues) > 0 {
+						chunk.update(column.Name.O, lowerValues[j], "")
 					}
-					if i != len(buckets)-1 {
-						upper = upperValues[j]
-						upperSymbol = lte
+				} else {
+					var lowerValue, upperValue string
+					if len(lowerValues) > 0 {
+						lowerValue = lowerValues[j]
 					}
-
-					chunk.update(col.Name.O, lower, lowerSymbol, upper, upperSymbol)
+					if len(upperValues) > 0 {
+						upperValue = upperValues[j]
+					}
+					chunk.update(column.Name.O, lowerValue, upperValue)
 				}
+			}
+
+			if i == len(buckets) {
+				chunks = append(chunks, chunk)
+				continue
+			}
 
+			count := (buckets[i].Count - latestCount) / int64(s.chunkSize)
+			if count == 0 {
+				continue
```

Comment:
if `buckets[i].Count - latestCount == 99` and `s.chunkSize == 100`, then `count == 0`, is a chunk need to be append for this?

Reply:
I think don't need, the chunk size is only a reference value, just make the rule simple